tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -2386,6 +2542,20 @@ void FragmentMetadata::load_generic_tile_offsets_v16_or_higher(
   gt_offsets_.tile_max_offsets_.resize(num);
   deserializer.read(&gt_offsets_.tile_max_offsets_[0], num * sizeof(uint64_t));
 
+  if (!dense_ &&
+      version_ >= constants::fragment_metadata_global_order_bounds_version) {
+    // Load offsets for the tile global order bounds
+    const auto num_dims = array_schema_->dim_num();
+    gt_offsets_.tile_global_order_min_offsets_.resize(num_dims);
+    gt_offsets_.tile_global_order_max_offsets_.resize(num_dims);
+    deserializer.read(
+        gt_offsets_.tile_global_order_min_offsets_.data(),
+        num_dims * sizeof(uint64_t));
+    deserializer.read(
+        gt_offsets_.tile_global_order_max_offsets_.data(),
+        num_dims * sizeof(uint64_t));
+  }
+
```

Comment:
Move to a new `load_generic_tile_offsets_v23_or_higher` function and remove the `version_` comparison?